typings/connection.d.ts

```diff
@@ -139,14 +141,24 @@ export interface ConnectionOptions extends EndpointOptions {
    * sender link on this connection. These options will be overridden by the specific sender options
    * that will be provided while creating a sender.
    */
-  sender_options?: SenderOptions
+  sender_options?: SenderOptions;
 
   /**
    * @property {ReceiverOptions} [receiver_options] Default options that can be provided while creating any
    * receiver link on this connection. These options will be overridden by the specific receiver options
    * that will be provided while creating a sender.
    */
-  receiver_options?: ReceiverOptions
+  receiver_options?: ReceiverOptions;
+  /**
+   * @property {Function} [connection_details] A function which if specified will be invoked to get the options
+   * to use (e.g. this can be used to alternate between a set of different host/port combinations)
+   */
+  connection_details?: Function;
```

Comment:
@grs - I would like to properly define the signature of connection_details. Can you help me define the signature?
I see a function named connection_details over [here](https://github.com/amqp/rhea/blob/master/lib/connection.js#L93) which takes in an options object (may be ConnectionOptions?) and returns an object which contains connect, host, port and the provided options as it's properties.

I also see connection_details taking in a number connection_counter over [here](https://github.com/amqp/rhea/blob/master/lib/connection.js#L227) and [here](https://github.com/amqp/rhea/blob/master/lib/connection.js#L236).

This makes it a little confusing. It would be nice if we can properly define the signature of the connection_details function w.r.t shape of the input and output.

Reply:
It takes a counter indicating the number of times the connection was established and returns an object to use as the (other) options to connect.

(The default is specified here: https://github.com/amqp/rhea/blob/master/lib/connection.js#L141)